### DISCUSSION TITLE
Advanced Foundry Section #1 (ERC20) question

### DISCUSSION THREAD

**Discussion by pierrealex3** [Cyfrin/foundry-full-course-cu#2428] [Q&A] [answered]
Hello,

There's a concept I don't get right I think...
In the test contract snippet below, in the setUp function, why do we prank msg.sender instead of address(deployer).
Since the OurToken contract was deployed by the DeployOurToken contract, aren't all the tokens minted to the DeployOurToken contract address??

```
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {Test} from "forge-std/Test.sol";
import {DeployOurToken} from "../script/DeployOurToken.s.sol";
import {OurToken} from "../src/OurToken.sol";

contract OurTokenTest is Test {
    OurToken public ourToken;
    DeployOurToken public deployer;

    address bob = makeAddr("bob");
    address alice = makeAddr("alice");

    uint256 public constant STARTING_BALANCE = 100 ether;

    function setUp() public {
        deployer = new DeployOurToken();
        ourToken = deployer.run();

        **vm.prank(msg.sender);**
        ourToken.transfer(bob, STARTING_BALANCE);
    }
}
```

**Comment by EngrPips** [comment, marked as answer]
Hello @pierrealex3, we had to prank `msg.sender` because in the deployment script, we wrapped the transaction that deployed the token contract between a `vm.startBroadcast()` and a `vm.stopBroadcast()` and that makes it so that it is not the deployment script that sends the transaction that deployed the token but any address associated with the private key that we pass into `vm.startBroadcast()` but if we did not pass in any private key then the transaction will be sent by the Foundry `Default-Sender` which is this address `0x1804c8AB1F12E6bbf3894d4083f33e07309d1f38`

**Reply by pierrealex3** [reply]
thanks, that makes it pretty clear!  Now I'm gonna have to get deeper in the topic of "how transactions are managed by foundry"

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.
